lib/src/components/NxSearchDropdown/stateful/__tests__/NxStatefulSearchDropdown.test.tsx

```diff
@@ -4,85 +4,349 @@
+    const el = quickRender({ defaultSearchText: 'foo' });
+    expect(el.getByRole('searchbox')).toHaveValue('foo');
   });
 
   it('passes the empty string as the searchText if defaultSearchText is not specified', function() {
-    expect(getShallow()).toHaveProp('searchText', '');
+    const el = quickRender();
+    expect(el.getByRole('searchbox')).toHaveValue('');
   });
 
-  it('updates the searchText based on values passed to the NxSearchDropdown onSearchTextChange callback', function() {
-    const component = getShallow();
+  it('sets the searchText as the value of the input', async function() {
+    const component = quickRender(),
+        input = component.getByRole('searchbox'),
+        user = userEvent.setup();
 
-    component.simulate('searchTextChange', 'foo');
-    expect(component).toHaveProp('searchText', 'foo');
+    await user.type(input, 'a');
 
-    component.simulate('searchTextChange', ' bar ');
-    expect(component).toHaveProp('searchText', ' bar ');
-  });
+    expect(input).toHaveValue('a');
 
-  it('passes its ref to the NxSearchDropdown', function() {
-    const ref = React.createRef<HTMLDivElement>();
-    const component = mount(<><NxStatefulSearchDropdown ref={ref} { ...minimalProps } /></>);
+    await user.clear(input);
+    await user.type(input, 'b');
 
-    expect(ref.current).toBeDefined();
-    expect(component.getDOMNode()).toBe(ref.current);
+    expect(input).toHaveValue('b');
   });
 
-  it('clears the searchText and fires its own onSelect when the NxSearchDropdown onSelect is fired', function() {
+  it('clears the searchText and fires its own onSelect when the dropdown\'s onSelect is fired', async function() {
     const onSelect = jest.fn(),
-        component = getShallow({ onSelect });
+        el = quickRender({ onSelect, matches: [{ id: 1, displayName: 'item 1' }] }),
+        input = el.getByRole('searchbox'),
+        user = userEvent.setup();
 
-    component.simulate('searchTextChange', '1');
-    expect(component).toHaveProp('searchText', '1');
+    await user.type(input, '1');
+    expect(input).toHaveValue('1');
     expect(onSelect).not.toHaveBeenCalled();
 
-    component.simulate('select', { id: 1, displayName: 'item 1' });
+    const menuItem = el.getByRole('menuitem', { name: 'item 1' });
+    await user.click(menuItem);
 
     expect(onSelect).toHaveBeenCalledWith({ id: 1, displayName: 'item 1' });
-    expect(component).toHaveProp('searchText', '');
+    expect(input).toHaveValue('');
+  });
+
+  it('calls onSearch whenever the input\'s onChange event fires with a value that differs after trimming, ' +
+  'passing the trimmed value', async function() {
+    const onSearch = jest.fn(),
+        el = quickRender({ defaultSearchText: 'foo ', onSearch }),
+        input = el.getByRole('searchbox');
+
+    expect(onSearch).not.toHaveBeenCalled();
+
+    fireEvent.change(input, { target: { value: 'foo' }});
```

Comment:
Don't forget to use `userEvent` here too

Reply:
> Don't forget to use `userEvent` here too

Yup, I've updated the stateful version to use `userEvent` here: aa5f41008258e8af27b6dc38fb812a28a69195f4